### ISSUE TITLE
[BUG] Gateway 504 after upgrade to 2.4.8

### ISSUE THREAD

**Issue by compumatter** [Leantime/leantime#2142]
**What set up are you using**
* [ ] Cloud Hosted Version
* [ X ] Self Hosted

**Describe the bug**
I've performed the upgrade of Leantime many times in the past.

This time after the upgrade:
* Upon getting to the url it prompted to update the database and proceeded without errors
* It presented the login as expected
* I logged in via LDAP and it allowed entrance as expected 
* I was able to browser around the interface with no problems.
* I created a new project called 'asdfasdf'
* I looked at the Todo options to see if anything had changed.
* I clicked on the "Todos" so I might add one.

That's where things went south and never came back.  From that point forward it took a reboot of the server environment just to be able to get to the login page.  That works consistently.  However as soon as I try to login, it now goes into the endless loop and finally a 504 Gateway Time-Out

I have verified that nginx is running, php8.1-fpm are running normally.

The only error I see of value anywhere is:
upstream timed out (110: Connection timed out) while reading response header from upstream, client: 192.168.10.44, server: projects.smvirtual.biz, request: "POST /auth/login HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock", host: "projects.smvirtual.biz", referrer: "https://projects.smvirtual.biz/auth/login

The sock file does exist with the correct permissions.

I am suspecting this issue has something to do with something in, or not in, the .env file but really don't know.  However it predictably loops until dead when attempting to login every time. 

**Leantime Version**
2.4.8

**Server**
Nginx

**PHP / MySQL Version**
php 8.1 Mariadb 10.4

**Additional context**
Youtube Video Demonstrating the problem:
* https://www.youtube.com/watch?v=APVfr7wr2Ks

Configuration Files below: (.env and nginx vhosts file)

# CONFIGURATION FILE (.env) with private items commented out with ******
```
# This is a sample configuration file with all possible configuration options.
# If you don't want to maintain a file like this you can pass in all variables via Server Variables

## Minimum Configuration, these are required for installation

LEAN_APP_URL = ''                                  # Base URL, only needed for subfolder installation
LEAN_APP_DIR = ''                                  # Base of application withotu trailing slash (used for cookies), e.g, /leantime

LEAN_DEBUG = 1                                     # Debug flag

# Database
LEAN_DB_HOST = 'localhost'                         # Database host
LEAN_DB_USER = '********'                # Database username
LEAN_DB_PASSWORD = '********'        # Database password
LEAN_DB_DATABASE = '********'            # Database name
LEAN_DB_PORT = '3306'                              # Database port


## Optional Configuraiton, you may ommit these from your .env file

## Default Settings
LEAN_SITENAME = 'ProjectMatter'                         # Name of your site, can be changed later
LEAN_LANGUAGE = 'en-US'                            # Default language
LEAN_DEFAULT_TIMEZONE = 'America/Los_Angeles'  # Set default timezone
LEAN_ENABLE_MENU_TYPE = false                      # Enable to specifiy menu on aproject by project basis
LEAN_SESSION_PASSWORD = '********'  #Salting sessions. Replace with a strong password
LEAN_SESSION_EXPIRATION = 32800                    # How many seconds after inactivity should we logout?  28800seconds = 8hours
LEAN_LOG_PATH = /var/log/sites-error.log    # Default Log Path (including filename), if not set /public_html/logs/error.log will be used
LEAN_DISABLE_LOGIN_FORM = false                    # If true then don't show the login form (useful only if additional auth method[s] are available)

## Look & Feel, these settings are available in the UI and can be overwritten there.
LEAN_LOGO_PATH = '/dist/images/logo.svg'           # Default logo path, can be changed later
LEAN_PRINT_LOGO_URL = '/dist/images/logo.png'      # Default logo URL use for printing (must be jpg or png format)
LEAN_DEFAULT_THEME = 'default'                     # Default theme
LEAN_PRIMARY_COLOR = '#1b75bb'                     # Primary Theme color
LEAN_SECONDARY_COLOR = '#81B1A8'                   # Secondary Theme Color
LEAN_KEEP_THEME = true

## Fileuploads

# Local File Uploads
LEAN_USER_FILE_PATH = 'userfiles/'                 # Local relative path to store uploaded files (if not using S3)
LEAN_DB_BACKUP_PATH = 'backupdb/'                  # Local relative path to store backup files, need permission to write

# S3 File Uploads
LEAN_USE_S3 = false                                # Set to true if you want to use S3 instead of local files
LEAN_S3_KEY = ''                                   # S3 Key
LEAN_S3_SECRET = ''                                # S3 Secret
LEAN_S3_BUCKET = ''                                # Your S3 bucket
LEAN_S3_USE_PATH_STYLE_ENDPOINT = false            # Sets the endpoint style: false => https://[bucket].[endpoint] ; true => https://[endpoint]/[bucket]
LEAN_S3_REGION = ''                                # S3 region
LEAN_S3_FOLDER_NAME = ''                           # Foldername within S3 (can be emtpy)
LEAN_S3_END_POINT = null                           # S3 EndPoint S3 Compatible (https://sfo2.digitaloceanspaces.com)

## Email
LEAN_EMAIL_RETURN = '********'        # Return email address, needs to be valid email address format
LEAN_EMAIL_USE_SMTP = false                        # Use SMTP? If set to false, the default php mail() function will be used
LEAN_EMAIL_SMTP_HOSTS = ''                         # SMTP host
LEAN_EMAIL_SMTP_AUTH = true                        # SMTP authentication required
LEAN_EMAIL_SMTP_USERNAME = ''                      # SMTP username
LEAN_EMAIL_SMTP_PASSWORD = ''                      # SMTP password
LEAN_EMAIL_SMTP_AUTO_TLS = true                    # SMTP Enable TLS encryption automatically if a server supports it
LEAN_EMAIL_SMTP_SECURE = ''                        # SMTP Security protocol (usually one of: TLS, SSL, STARTTLS)
LEAN_EMAIL_SMTP_SSLNOVERIFY = false                # SMTP Allow insecure SSL: Don't verify certificate, accept self-signed, etc.
LEAN_EMAIL_SMTP_PORT = ''                          # Port (usually one of 25, 465, 587, 2526)

## Ldap
LEAN_LDAP_USE_LDAP = true                         # Set to true if you want to use LDAP
# new 2023.12.12
LEAN_LDAP_LDAP_DOMAIN = '********' # Domain name after username@ so users can login without domain definition
LEAN_LDAP_LDAP_TYPE = 'OL'                         # Select the correct directory type. Currently Supported: OL - OpenLdap, AD - Active Directory
LEAN_LDAP_HOST = 'localhost'                       # FQDN
LEAN_LDAP_PORT = 389                               # Default Port
# new 2023.12.12
LEAN_LDAP_URI = ''                                 # LDAP URI as alternative to hostname and port. Uses ldap://hostname:port
LEAN_LDAP_DN = 'cn=********,dc=********,dc=********,dc=biz'  # Location of users, example: CN=users,DC=example,DC=com

LEAN_LDAP_KEYS="{
        \"username\":\"cn\",
        \"groups\":\"memberof\",
        \"email\":\"mail\",
        \"firstname\":\"givenName\",
        \"lastname\":\"sn\",
        \"phone\":\"telephoneNumber\",
        \"jobTitle\":\"title\",
        \"jobLevel\":\"level\",
        \"department\":\"department\"

}"

# For AD use these default attributes
# LEAN_LDAP_KEYS="{
#        \"username\":\"cn\",
#        \"groups\":\"memberOf\",
#        \"email\":\"mail\",
#        \"firstname\":\"givenName\",
#        \"lastname\":\"sn\",
#        \"phone\":\"telephoneNumber\",
#        \"jobTitle\":\"title\"
#        \"jobLevel\":\"level\"
#        \"department\":\"department\"
#      }"

LEAN_LDAP_DEFAULT_ROLE_KEY = 20;   # Default Leantime Role on creation. (set to editor)

# Default role assignments upon first login.
# (Optional) Can be updated later in user settings for each user
LEAN_LDAP_GROUP_ASSIGNMENT="{
               \"5\": {
                 \"ltRole\":\"readonly\",
                 \"ldapRole\":\"proj-m-r\"
               },
               \"10\": {
                 \"ltRole\":\"commenter\",
                  \"ldapRole\":\"proj-m-c\"
               },
               \"20\": {
                 \"ltRole\":\"editor\",
                  \"ldapRole\":\"proj-m-e\"
               },
               \"30\": {
                 \"ltRole\":\"manager\",
                  \"ldapRole\":\"proj-m-m\"
               },
               \"40\": {
                 \"ltRole\":\"admin\",
                  \"ldapRole\":\"proj-m-a\"
               },
               \"50\": {
                 \"ltRole\":\"owner\",
                 \"ldapRole\":\"administators\"
               }
}"

## OpenID Connect
# required
LEAN_OIDC_ENABLE = false
LEAN_OIDC_CLIENT_ID =
LEAN_OIDC_CLIENT_SECRET =

# required - the URL for your provider (examples down below)
#LEAN_OIDC_PROVIDER_URL =

#Create User if it doesn't exist in Leantime db, otherwise fail login
LEAN_OIDC_CREATE_USER = false

# Default role for users created via OIDC (20 is editor)
LEAN_OIDC_DEFAULT_ROLE = 20

# optional - these will be read from the well-known configuration if possible
#LEAN_OIDC_AUTH_URL_OVERRIDE =
#LEAN_OIDC_TOKEN_URL_OVERRIDE =
#LEAN_OIDC_JWKS_URL_OVERRIDE =
#LEAN_OIDC_USERINFO_URL_OVERRIDE =

# optional - override the public key for RSA validation
#LEAN_OIDC_CERTIFICATE_STRING =
#LEAN_OIDC_CERTIFICATE_FILE =

# optional - override the requested scopes
#LEAN_OIDC_SCOPES =

# optional - override the keys used for these fields
#LEAN_OIDC_FIELD_EMAIL =
#LEAN_OIDC_FIELD_FIRSTNAME =
#LEAN_OIDC_FIELD_LASTNAME =
#LEAN_OIDC_FIELD_PHONE =
#LEAN_OIDC_FIELD_JOBTITLE =
#LEAN_OIDC_FIELD_JOBLEVEL=
#LEAN_OIDC_FIELD_DEPARTMENT =

## OpenID Connect setting for GitHub
#LEAN_OIDC_PROVIDER_URL = https://token.actions.githubusercontent.com/
#LEAN_OIDC_AUTH_URL_OVERRIDE = https://github.com/login/oauth/authorize
#LEAN_OIDC_TOKEN_URL_OVERRIDE = https://github.com/login/oauth/access_token
#LEAN_OIDC_USERINFO_URL_OVERRIDE = https://api.github.com/user,https://api.github.com/user/emails
#LEAN_OIDC_SCOPES = user:email,read:user
#LEAN_OIDC_FIELD_EMAIL = 0.email
#LEAN_OIDC_FIELD_FIRSTNAME = name

## Redis (for session storage and cache)
LEAN_USE_REDIS = false          #Set to true to use redis as session cache
LEAN_REDIS_URL = ''             #Add url path such as tcp://1.2.3.4:6379. If you are using a password add ?auth=yourverycomplexpasswordhere to your url
```

NGINX CONFIGURATION FILE: I have compared to your suggested online and it looks proper 
```
server {
  listen 80;
  server_name projects.smvirtual.biz;
  return 301 https://projects.smvirtual.biz$request_uri;
}

server {
  listen 443 ssl http2;
  server_name projects.smvirtual.biz;
  set $base /SM_DATA/web_projects/public_html;
  root $base/public;
  include sm-snippets/wildcard_ssl_certs;
# logging
  include sm-snippets/log;
  # index.php
  index index.php;

  location ~.php$ {

    # 404
    try_files $fastcgi_script_name =404;

    # default fastcgi_params
    include fastcgi_params;

    # fastcgi settings
    fastcgi_pass unix:/var/run/php/php8.1-fpm.sock;
    fastcgi_index index.php;
    fastcgi_buffers 8 16k;
    fastcgi_buffer_size 32k;

    # fastcgi params
    fastcgi_param DOCUMENT_ROOT $realpath_root;
    fastcgi_param SCRIPT_FILENAME       $realpath_root$fastcgi_script_name;
    fastcgi_param PHP_ADMIN_VALUE       "open_basedir=$base/:/usr/lib/php/:/tmp/";
  }

  location / {
    try_files $uri $uri/ /index.php?$args;
  }


  # additional config
  # favicon.ico
  location = /favicon.ico {
    log_not_found off;
    access_log off;
  }

  # robots.txt
  location = /robots.txt {
    log_not_found off;
    access_log off;
  }

  # assets, media
  location ~* \.(?:css(\.map)?|js(\.map)?|jpe?g|png|gif|ico|cur|heic|webp|tiff?|mp3|m4a|aac|ogg|midi?|wav|mp4|mov|webm|mpe?g|avi|ogv|flv|wmv)$ {
    expires 7d;
    access_log off;
  }

  # svg, fonts
  location ~* \.(?:svgz?|ttf|ttc|otf|eot|woff2?)$ {
    add_header Access-Control-Allow-Origin "*";
    expires 7d;
    access_log off;
  }

  # gzip
  gzip on;
  gzip_vary on;
  gzip_proxied any;
  gzip_comp_level 6;
  gzip_types text/plain text/css text/xml application/json application/javascript application/rss+xml application/atom+xml image/svg+xml;
}

```

**Comment by marcelfolaron** [comment]
Have you seen any errors in your error.log file?

**Comment by compumatter** [comment]
Hello Marcel.  I am finally able to get back to this important issue.  This is what I see in the error log:

Upon the initial loading of the leantime login page I see this error in the error log:

`2024/01/08 20:11:16 [error] 2861041#2861041: *1 FastCGI sent in stderr: "PHP message: PHP Warning:  ini_set(): open_basedir restriction in effect. File(/var/log/sites-error.log) is not within the allowed path(s): (/SM_DATA/web_projects/public_html/:/usr/lib/php/:/tmp/) in /SM_DATA/web_projects/public_html/app/Core/AppSettings.php on line 66" while reading response header from upstream, client: 67.168.11.91, server: projects.smvirtual.biz, request: "GET /api/i18n HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "projects.smvirtual.biz", referrer: "https://projects.smvirtual.biz/auth/login"`

Upon the failing of the login with a 504 error I see this error in the error log.

`2024/01/08 20:11:59 [error] 2861041#2861041: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 67.168.11.91, server: projects.smvirtual.biz, request: "POST /auth/login HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "projects.smvirtual.biz", referrer: "https://projects.smvirtual.biz/auth/login"`

Restarting php8.1-fpm allows me to reach the login page again.  If I don't do that, it will not allow me to get to the login page again.

I look forward to your thoughts.

Jay

**Comment by compumatter** [comment]
I have reinstalled php-fpm 8.1 to no avail.

I was able to remove the first error "on page load" by commenting this out in the virtualhosts file:
# fastcgi_param PHP_ADMIN_VALUE      "open_basedir=$base/:/usr/lib/php/:/tmp/";

However the road block error which results in a 504 error after a long time delay when pressing submit after login I have not resolved.
`2024/01/08 20:11:59 [error] 2861041#2861041: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 67.168.11.91, server: projects.smvirtual.biz, request: "POST /auth/login HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "projects.smvirtual.biz", referrer: "https://projects.smvirtual.biz/auth/login"`

Some of the things I have done.

I have increased these values / restarted all services (php-fpm* and nginx) based on forum posts elsewhere to no avail:
/etc/php/8.1/fpm/pool.d/www.conf
pm.max_children from 5 to 10
pm.start_servers from 2 to 4
pm.min_spare_servers from 1 to 2
pm.min_spare_servers from 3 to 6

I have increased these timeout values in my virtualhosts file for nginx to no effect other than it takes longer to fail.
proxy_connect_timeout 180s;
proxy_send_timeout 180s;
proxy_read_timeout 180s;
fastcgi_send_timeout 180s;
fastcgi_read_timeout 180s;

I have modified buffer sizes in the virtual hosts file as talked about at this url:
proxy_busy_buffers_size   512k;
proxy_buffers   4 512k;
proxy_buffer_size   256k;
 
Still no love - same symptom.

**Comment by marcelfolaron** [comment]
Hi Jay, 
The first logs that you sent say "open_basedir restriction in effect. File(/var/log/sites-error.log) " This indicates that the folder you are using is not within the allowed paths of your fpm. In this particular case though it's your definition for the error_log file. Since your base is set to: /SM_DATA/web_projects/public_html it needs to live wihtin that folder.

**Comment by compumatter** [comment]
Hello Marcel,

The open_basedir restriction error was cured by commenting out that line in the vhosts file.

However the problem with logging in and the 504 continued to exist.

I have solved this problem and it was no fault of Leantime.

There was an unrelated error appearing in the php8.1-fpm log files that showed:
`postfix/postdrop[710129]: warning: mail_queue_enter: create file maildrop/292754.710129: Permission denied`

Because it was not showing any connection to Leantime or my vhost for that domain I thought it was unrelated and did not connect the dots that one might be preventing the other.

However, once permissions were fixed on that postfix environment the Leantime logins worked normally :-)

My sincere apologies for creating an issue of this.  I did not imagine such a connection would cause a problem with Leantime that did not reveal itself in other ways on the server.

Sincerely,

Jay
